filter_spatial.json

```diff
@@ -1,7 +1,7 @@
 {
-    "id": "filter_polygon",
-    "summary": "Spatial filter using polygons",
-    "description": "Limits the data cube over the spatial dimensions to the specified polygons.\n\nThe filter retains a pixel in the data cube if the point at the pixel center intersects with at least one of the polygons (as defined in the Simple Features standard by the OGC).",
+    "id": "filter_spatial",
+    "summary": "Spatial filter using geometries",
+    "description": "Limits the data cube over the spatial dimensions to the specified geometries.\n\n- For **polygons**, the filter retains a pixel in the data cube if the point at the pixel center intersects with at least one of the polygons (as defined in the Simple Features standard by the OGC).\n- For **points** and **lines** (line strings), the filter retains all the pixel centers closest to any of the geometries.",
```

Comment:
What happens to pixels outside of the geometries but still inside the corresponding bbox? Are they set to null?
If so: what is the difference with mask_polygon?
If not: what is the difference with filter_bbox?

Reply:
They get removed whenever possible and all the other get set to null. It's basically like applying filter_bbox with the bbox of the geometry and all the other values not covered by the geometry in the bbox get set to null. Mask polygon doesn't restrict the area and you can choose replacement value and invert the area.

Reply:
maybe too subtle to justify defining separate processes? For example, we could add an option like "optimize_bbox" to the mask process to implement what filter_spatial is doing.
To me, "mask_polygon" sounds more self-explanatory than "filter_spatial" anyway.

Reply:
I think both have their value. There's another difference: mask_polygon doesn't support points and lines, which I think is somewhat reasonable.